### PR TITLE
[Internal] Client Telemetry: Fixes loggers

### LINKED ISSUE #4592: [Client Telemetry] - Trace error not recording exception
The index needs to be 0 instead of 1. This results in the trace messages not logging the exception. 

https://github.com/Azure/azure-cosmos-dotnet-v3/blob/f277ce3428798ec5d90e1a6401f605e4e0cdb957/Microsoft.Azure.Cosmos/src/Telemetry/Collector/TelemetryCollector.cs#L66

Example: DocDBTrace Error: 0 : Error while collecting operation telemetry. Exception : 1

### LINKED ISSUE #4593: [Client Telemetry] - Index out of range
The request latency can sometimes be less than 0 in the storeStatistics.RequestLatency.Ticks resulting in the following error.

```
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at HdrHistogram.Utilities.Bitwise.Log2(Int32 i) in C:\\azure-cosmos-dotnet-v3\\Microsoft.Azure.Cosmos\\src\\OSS\\HdrHistogram\\Utilities\\Bitwise.cs:line 87\r
   at HdrHistogram.Utilities.Bitwise.NumberOfLeadingZeros(Int64 value) in C:\\azure-cosmos-dotnet-v3\\Microsoft.Azure.Cosmos\\src\\OSS\\HdrHistogram\\Utilities\\Bitwise.cs:line 58\r
   at HdrHistogram.HistogramBase.GetBucketIndex(Int64 value, Int64 subBucketMask, Int32 bucketIndexOffset) in C:\\azure-cosmos-dotnet-v3\\Microsoft.Azure.Cosmos\\src\\OSS\\HdrHistogram\\HistogramBase.cs:line 696\r
   at HdrHistogram.HistogramBase.GetBucketIndex(Int64 value) in C:\\azure-cosmos-dotnet-v3\\Microsoft.Azure.Cosmos\\src\\OSS\\HdrHistogram\\HistogramBase.cs:line 692\r
   at HdrHistogram.HistogramBase.RecordSingleValue(Int64 value) in C:\\azure-cosmos-dotnet-v3\\Microsoft.Azure.Cosmos\\src\\OSS\\HdrHistogram\\HistogramBase.cs:line 634\r
   at HdrHistogram.HistogramBase.RecordValu
e(Int64 value) in C:\\azure-cosmos-dotnet-v3\\Microsoft.Azure.Cosmos\\src\\OSS\\HdrHistogram\\HistogramBase.cs:line 238\r
   at Microsoft.Azure.Cosmos.Telemetry.NetworkDataRecorder.Record(List`1 storeResponseStatistics, String databaseId, String containerId) in C:\\azure-cosmos-dotnet-v3\\Microsoft.Azure.Cosmos\\src\\Telemetry\\NetworkDataRecorder.cs:line 42\r
   at Microsoft.Azure.Cosmos.Telemetry.ClientTelemetry.PushNetworkDataPoint(List`1 storeResponseStatistics, String databaseId, String containerId) in C:\\azure-cosmos-dotnet-v3\\Microsoft.Azure.Cosmos\\src\\Telemetry\\ClientTelemetry.cs:line 348\r
   at Microsoft.Azure.Cosmos.Telemetry.Collector.TelemetryCollector.CollectOperationAndNetworkInfo(Func`1 functionFordata) in C:\\azure-cosmos-dotnet-v3\\Microsoft.Azure.Cosmos\\src\\Telemetry\\Collector\\TelemetryCollector.cs:line 61"
```
storeStatistics.RequestLatency.Ticks = -20355

```
storeStatistics
{Microsoft.Azure.Cosmos.Tracing.TraceData.ClientSideRequestStatisticsTraceDatum.StoreResponseStatistics}
    IsSupplementalResponse: true
    LocationEndpoint: {https://jake-ppaf-61-716-northcentralus.documents-test.windows-int.net/}
    RequestLatency: {-00:00:00.0020355}
    RequestOperationType: Head
    RequestResourceType: Collection
    RequestResponseTime: {7/19/2024 8:21:51 PM}
    RequestSessionToken: null
    RequestStartTime: {7/19/2024 8:21:51 PM}
    StoreResult: {StorePhysicalAddress: rntbd://cdb-ms--northcentralus1-be1.documents.windows-int.net:14300/apps/f84da795-09c0-4f07-8a16-88be58a74766/services/445a8bcb-1cb4-42a4-95bd-e01111111/partitions/6441ce84-43b3-4b8e-93d0-edde1111111/replicas/133634132488190899s/, LSN: 153713, GlobalCommittedLsn: 153713, PartitionKeyRangeId: 1, IsValid: True, StatusCode: 204, SubStatusCode: 0, RequestCharge: 0, ItemLSN: 1, SessionToken: -1#153713, UsingLocalLSN: False, TransportException: null, BELatencyMs: 0.243, ActivityId: e9dbcf03-2efa-4e4d-8a87-5e61932165e9, RetryAfterInMs: , ReplicaHealthStatuses: [(port: 14301 | status: Unknown | lkt: 7/19/2024 8:21:48 PM),(port: 14003 | status: Unknown | lkt: 7/19/2024 8:21:48 PM),(port: 14300 | status: Unknown | lkt: 7/19/2024 8:21:48 PM),(port: 14302 | status: Unknown | lkt: 7/19/2024 8:21:48 PM)], TransportRequestTimeline: {"requestTimeline":[{"event": "Created", "startTimeUtc": "2024-07-19T20:21:51.3250888Z", "durationInMs": 0.0079},{"event": "ChannelAcquisitio
nStarted", "startTimeUtc": "2024-07-19T20:21:51.3250967Z", "durationInMs": 0.0068},{"event": "Pipelined", "startTimeUtc": "2024-07-19T20:21:51.3251035Z", "durationInMs": 2.2707},{"event": "Transit Time", "startTimeUtc": "2024-07-19T20:21:51.3273742Z", "durationInMs": 20.8275},{"event": "Received", "startTimeUtc": "2024-07-19T20:21:51.3482017Z", "durationInMs": 1.8793},{"event": "Completed", "startTimeUtc": "2024-07-19T20:21:51.3500810Z", "durationInMs": 0}],"serviceEndpointStats":{"inflightRequests":1,"openConnections":1},"connectionStats":{"waitforConnectionInit":"False","callsPendingReceive":0,"lastSendAttempt":"2024-07-19T20:21:51.1875266Z","lastSend":"2024-07-19T20:21:51.1879217Z","lastReceive":"2024-07-19T20:21:51.2471996Z"},"requestSizeInBytes":410,"responseMetadataSizeInBytes":254};}
```

### PR DESCRIPTION
**Pull request by sourabh1007**
## Description

This PR would fix this issue: https://github.com/Azure/azure-cosmos-dotnet-v3/issues/4592

## Type of change
- [] Bug fix (non-breaking change which fixes an issue)

## Closing issues

To automatically close an issue: closes #4592

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Telemetry/Collector/TelemetryCollector.cs
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Telemetry/Collector/TelemetryCollector.cs:1]: remove $

File: Microsoft.Azure.Cosmos/src/Telemetry/NetworkDataRecorder.cs [hunk 1]
```diff
@@ -24,6 +25,12 @@ public void Record(List<StoreResponseStatistics> storeResponseStatistics, string
             {
                 if (NetworkDataRecorder.IsStatusCodeNotExcluded((int)storeStatistics.StoreResult.StatusCode, (int)storeStatistics.StoreResult.SubStatusCode))
                 {
+                    if (storeStatistics.RequestLatency.Ticks < 0)
+                    {
+                        DefaultTrace.TraceWarning($"Skipping invalid latency value i.e.{0} ", storeStatistics.RequestLatency.Ticks);
```
Comment (bartelink) [inline Microsoft.Azure.Cosmos/src/Telemetry/NetworkDataRecorder.cs:30]:
```suggestion
                        DefaultTrace.TraceWarning("Skipping invalid latency value {0}", storeStatistics.RequestLatency.Ticks);
```

**Review by NaluTripician — APPROVED** [review]

**Review by philipthomas-MSFT — APPROVED** [review]
LGTM

**Review by kirankumarkolli — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/Telemetry/Collector/TelemetryCollector.cs [hunk 2]
```diff
@@ -42,7 +42,7 @@ public void CollectCacheInfo(string cacheName, Func<TelemetryInformation> functi
             }
             catch (Exception ex)
             {
-                DefaultTrace.TraceError($"Error while collecting cache {0} telemetry. Exception : {1}", cacheName, ex);
+                DefaultTrace.TraceError("Error while collecting cache {0} telemetry. Exception : {1}", cacheName, ex);
```
Comment (bartelink) [inline Microsoft.Azure.Cosmos/src/Telemetry/Collector/TelemetryCollector.cs:45]: if there's an overload that takes the ex arg first, that should be used instead (same for others)

**Pull request merged**